pytest_bdd/parser.py

```diff
@@ -281,11 +284,45 @@ def validate(self):
         """
         params = self.params
         example_params = self.get_example_params()
-        if params and example_params and params != example_params:
+        if params or example_params:
+            self._validate_example_free_variables()
+            self._validate_step_free_variables()
+
+    def _validate_example_free_variables(self):
+        params = self.params
+        example_params = self.get_example_params()
+        if self.allow_example_free_variables or example_params.issubset(params):
+            return
+        else:
+            raise exceptions.ScenarioExamplesNotValidError(
+                (
+                    """Scenario "{}" in the feature "{}" does not have valid examples. """
+                    """Set of example parameters {} should be a subset of step """
+                    """parameters {} if examples open variables are not allowed"""
```

Comment:
Are we calling them "open" variables, or "free" variables? Does either term have a clear and obvious meaning?